src/attribute_type.rs

```diff
@@ -2496,24 +2498,24 @@ fn parse_rtcp(to_parse: &str) -> Result<SdpAttribute, SdpParserInternalError> {
                     ));
                 }
                 Some(x) => {
-                    let addrtype = parse_address_type(x)?;
+                    let addrtype = AddressType::from_str(x)?;
                     let addr = match tokens.next() {
                         None => {
                             return Err(SdpParserInternalError::Generic(
                                 "Rtcp attribute is missing ip address token".to_string(),
                             ));
                         }
-                        Some(x) => {
-                            let addr = parse_unicast_address(x)?;
-                            if !addrtype.same_protocol(&addr) {
+                        Some(x) => match ExplicitlyTypedAddress::try_from((addrtype, x)) {
+                            Ok(address) => address,
+                            Err(SdpParserInternalError::AddressTypeMismatch) => {
                                 return Err(SdpParserInternalError::Generic(
                                     "Failed to parse unicast address attribute.\
                                      addrtype does not match address."
                                         .to_string(),
                                 ));
                             }
```

Comment:
Isn't here a comma missing?

Reply:
I guess clippy doesn't care about that? I will fix it.

Reply:
Fixed